Realm/ObjectStore/object_store.cpp

```diff
@@ -0,0 +1,414 @@
+        auto target_prop = target_schema.property_for_name(current_prop.name);
+
+        if (!target_prop) {
+            validation_errors.push_back("Property '" + current_prop.name + "' is missing from latest object model.");
+            continue;
+        }
+
+        if (current_prop.type != target_prop->type) {
+            validation_errors.push_back("Property types for '" + target_prop->name + "' property do not match. " +
+                                        "Old type '" + string_for_property_type(current_prop.type) +
+                                        "', new type '" + string_for_property_type(target_prop->type) + "'");
+            continue;
+        }
+        if (current_prop.type == PropertyTypeObject || target_prop->type == PropertyTypeArray) {
+            if (current_prop.object_type != target_prop->object_type) {
+                validation_errors.push_back("Target object type for property '" + current_prop.name + "' does not match. " +
+                                            "Old type '" + current_prop.object_type +
+                                            "', new type '" + target_prop->object_type + "'.");
+            }
+        }
+        if (current_prop.is_primary != target_prop->is_primary) {
+            if (current_prop.is_primary) {
+                validation_errors.push_back("Property '" + current_prop.name + "' is no longer a primary key.");
+            }
+            else {
+                validation_errors.push_back("Property '" + current_prop.name + "' has been made a primary key.");
+            }
+        }
+
+        // create new property with aligned column
+        target_prop->table_column = current_prop.table_column;
+    }
+
+    // check for new missing properties
+    for (auto& target_prop : target_schema.properties) {
+        if (!table_schema.property_for_name(target_prop.name)) {
+            validation_errors.push_back("Property '" + target_prop.name + "' has been added to latest object model.");
+        }
+    }
+
+    return validation_errors;
+}
+
+void ObjectStore::update_column_mapping(Group *group, ObjectSchema &target_schema) {
+    ObjectSchema table_schema(group, target_schema.name);
+    for (auto& target_prop : target_schema.properties) {
+        auto table_prop = table_schema.property_for_name(target_prop.name);
+        REALM_ASSERT_DEBUG(table_prop);
+
+        target_prop.table_column = table_prop->table_column;
+    }
+}
+
+static inline bool property_has_changed(Property &p1, Property &p2) {
+    return p1.type != p2.type || p1.name != p2.name || p1.object_type != p2.object_type;
+}
+
+// set references to tables on targetSchema and create/update any missing or out-of-date tables
+// if update existing is true, updates existing tables, otherwise validates existing tables
+// NOTE: must be called from within write transaction
+bool ObjectStore::create_tables(Group *group, ObjectStore::Schema &target_schema, bool update_existing) {
```

Comment:
This method is a good role model example of commenting. I really appreciate the short but precise one-line comments ahead of a small code section. :+1:  Some of the other methods could be more like this :-)